### PR TITLE
Improve Date/DateTime type parameters

### PR DESCRIPTION
**Pull request by kravets-levko**
Issue getredash/redash#2642

Default date/time inputs replaced with [Ant Date/Time picker](https://ant.design/components/date-picker/).

**Screenshots:**

Date:
image
image

Date/Time:
image
image
image

Date/Time with seconds (same as Date/Time but different format):
image

### CODE REVIEW

File: client/app/services/query.js [hunk 1]
```diff
@@ -28,6 +28,12 @@ class Parameter {
     this.global = parameter.global;
     this.enumOptions = parameter.enumOptions;
     this.queryId = parameter.queryId;
+
+    // method to update parameter value from date/time picker component
+    // (react does not support two-way binding with `ngModel`)
+    this.updateValue = (function updateValue(value) {
+      this.ngModel = value;
+    }).bind(this);
```
[the remark this answers is not in the document]
Comment (kravets-levko) [reply client/app/services/query.js]: oh, missed it, you're right

**Comment by kravets-levko** [comment]
@arikfr Updated PR.

> how does it look side by side with regular input fields?

A bit different. I'm not sure - should I fix the CSS, or it's better if @kocsmy will do it? WDYT?
image

**Comment by kravets-levko** [comment]
> Also try to "stress" test it with a few edge case. I remember the browser default input field had issues where it would show a value but actually have a null value.

I tried to use it from users perspective - and it works as it should: it displays value if there is value set, and it shows placeholder when param does not have value set.

**Comment by arikfr** [comment]
> A bit different. I'm not sure - should I fix the CSS, or it's better if @kocsmy will do it? WDYT?

There are 3 options here:

1. Change the Ant component to look like ours (or at least align properly).
2. Change ours to look like Ant.
3. Use [Ant's Input component](https://ant.design/components/input/). 

@kocsmy what are your thoughts?

**Comment by kocsmy** [comment]
Hmm, it should be fairly simple to change their design to look like ours but if we plan to use more of Ant components or even use their whole library, then that's a bigger plan :)

**Comment by arikfr** [comment]
@kocsmy 

> Hmm, it should be fairly simple to change their design to look like ours but if we plan to use more of Ant components or even use their whole library, then that's a bigger plan :)

We will use more and more of their components to the point where we can say that we use the whole lib :)

I merged this already so give this a look and see what's the simplest thing we can do to make it look less alien, before we commit to a bigger plan.

**Comment by kocsmy** [comment]
Sure, I'll take a look and see if we can improve this quickly.
